From 5fcd012c0fc31ddeb4ff09a3f3d64ae5fcd03c0a Mon Sep 17 00:00:00 2001
From: Sauhard Pande <sauhardp@codeaurora.org>
Date: Fri, 14 Apr 2017 03:17:06 +0530
Subject: [PATCH 3/3] Camera: HAL1 fix for crash in releaserecording frame

Issue:
During stoprecording deallocate meta is called which
erases the entry from map in CameraDevice and also
releases metadatabuffer. Thus when releaseRecordingframe
is called the entry is not found in the map causing crash
due to null pointer reference

Fix:
If there is no entry for that buffer in map then return

Change-Id: I937754bbc68bdf2fd54e22a1ffc674f90730fb37
---
 camera/device/1.0/default/CameraDevice.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/camera/device/1.0/default/CameraDevice.cpp b/camera/device/1.0/default/CameraDevice.cpp
index aa57db98..50cecd77 100644
--- a/camera/device/1.0/default/CameraDevice.cpp
+++ b/camera/device/1.0/default/CameraDevice.cpp
@@ -881,6 +881,10 @@ void CameraDevice::releaseRecordingFrameLocked(
         return;
     }
     if (mDevice->ops->release_recording_frame) {
+        if (mMemoryMap.count(memId) == 0) {
+            ALOGE("Buffer with id %d already released", memId);
+            return;
+        }
         CameraHeapMemory* camMemory;
         {
             Mutex::Autolock _l(mMemoryMapLock);
-- 
2.11.0.windows.1

